backends/arm/test/ops/test_linear.py

```diff
@@ -0,0 +1,232 @@
+        30,
+    ),
+]
+
+
+class TestLinear(unittest.TestCase):
+    class Linear(torch.nn.Module):
+        def __init__(
+            self,
+            in_features: int,
+            out_features: int = 3,
+            bias: bool = True,
+        ):
+            super().__init__()
+            self.fc = torch.nn.Linear(
+                in_features=in_features,
+                out_features=out_features,
+                bias=bias,
+            )
+
+        def forward(self, x):
+            return self.fc(x)
+
+    def _test_linear_tosa_MI_pipeline(
+        self, module: torch.nn.Module, test_data: Tuple[torch.Tensor]
+    ):
+        tester = (
+            ArmTester(
+                module,
+                inputs=test_data,
+                profile=TosaProfile.MI,
+                backend=ArmBackendSelector.TOSA,
+            )
+            .export()
+            .check_count({"torch.ops.aten.addmm.default": 1})
+            .check_not(["torch.ops.quantized_decomposed"])
+            .to_edge()
+            .partition()
+            .check_count({"torch.ops.higher_order.executorch_call_delegate": 1})
+            .to_executorch()
+        )
+        if TOSA_REF_MODEL_INSTALLED:
+            tester.run_method().compare_outputs()
+        else:
+            logger.warning(
+                "TOSA ref model tool not installed, skip numerical correctness tests"
+            )
+
+    def _test_linear_tosa_BI_pipeline(
+        self, module: torch.nn.Module, test_data: Tuple[torch.Tensor]
+    ):
+        tester = (
+            ArmTester(
+                module,
+                inputs=test_data,
+                profile=TosaProfile.BI,
+                backend=ArmBackendSelector.TOSA,
+            )
+            .quantize()
+            .export()
+            .check_count({"torch.ops.aten.addmm.default": 1})
+            .check(["torch.ops.quantized_decomposed"])
+            .to_edge()
+            .partition()
+            .check_count({"torch.ops.higher_order.executorch_call_delegate": 1})
+            .to_executorch()
+        )
+        if TOSA_REF_MODEL_INSTALLED:
+            tester.run_method().compare_outputs(qtol=True)
+        else:
+            logger.warning(
+                "TOSA ref model tool not installed, skip numerical correctness tests"
+            )
+
+    def _test_linear_tosa_u55_BI_pipeline(
+        self, module: torch.nn.Module, test_data: Tuple[torch.Tensor]
+    ):
+        (
+            ArmTester(
+                module,
+                inputs=test_data,
+                profile=TosaProfile.BI,
+                backend=ArmBackendSelector.ETHOS_U55,
+            )
+            .quantize()
+            .export()
+            .check_count({"torch.ops.aten.addmm.default": 1})
+            .check(["torch.ops.quantized_decomposed"])
+            .to_edge()
+            .partition()
+            .check_count({"torch.ops.higher_order.executorch_call_delegate": 1})
+            .to_executorch()
```

Comment:
add the optional compare using FVP? or too convoluted for a unit test

Reply:
Atm we're not planning to test that in this flow, although it's an interesting idea. Challenges:
1. FVP is currently not available for all OS's
2. We need to build the runtime
This might be something for the future though...